### DISCUSSION TITLE
Stripe Solito

### DISCUSSION THREAD

**Discussion by karmatradeDev** [nandorojo/solito#107] [Q&A]
Hey quick question anyone use stripe with solito yet? I'm familiar with stripe but when looking at the expo doc https://docs.expo.dev/versions/latest/sdk/stripe/ it says doesn't support web. so before i make 2 separate solutions i wanted to see if anyone has made a cross-platform solution

**Comment by react-ghost** [comment]
Interested in this as well.

**Comment by luizuk** [comment]
I'm interested in this too! After my research and implementation try, I will share it here.

**Comment by khevamann** [comment]
## Gathering Payment Methods
I have looked into this quite a bit and the best approach from what I have found (assuming you want the highest tier of PCI compliance provided) is to use the payment elements from the react native SDK on mobile and stripe elements on web. Then handling as much as possible server-side. https://stripe.com/docs/payments/elements

So the cleanest abstraction would be to add the elements as custom components to gather payment methods then pass that to your server to handle the payment.

So for adding a credit card input you could use the respective methods to make a form for web and mobile:
CreditCardForm.web.ts
CreditCardForm.ts

Or same if you want payment sheet/ mobile wallet payment options. Then with the source returned from Stripe create the payment on your server.

If you are not worried about meeting the highest PCI compliance level you can simply use the stripe client API and build your own credit card form which can be shared on mobile and web. https://stripe.com/docs/api/tokens/create_card

```js
const token = await stripe.tokens.create({
  card: {
    number: '4242424242424242',
    exp_month: 9,
    exp_year: 2023,
    cvc: '314',
  },
});
```

More about Stripe PCI compliance here: https://stripe.com/docs/security/guide#validating-pci-compliance
